kernel/src/transaction.rs

```diff
@@ -54,8 +56,7 @@ pub fn get_write_metadata_schema() -> &'static SchemaRef {
 /// ```
 pub struct Transaction {
     read_snapshot: Arc<Snapshot>,
-    operation: Option<String>,
-    commit_info: Option<Arc<dyn EngineData>>,
+    commit_info: Option<Box<CommitInfo>>,
```

Comment:
I wonder if (for now) we should just keep `operation` and `engine_commit_info` hashmap?